src/main/java/org/helpberkeley/memberdata/Main.java

```diff
@@ -743,13 +757,25 @@ private static void doUpdateMemberData(
     }
 
     private static void doOneKitchenDriverMessages(
-            ApiClient apiClient, WorkRequestHandler.WorkRequest request, Map<String, User> users) {
+            ApiClient apiClient, WorkRequestHandler.WorkRequest request, Map<String, User> users, String posterUsername) {
 
-        Topic topic = (request.destinationTopic != null) ? request.destinationTopic : Constants.TOPIC_DRIVERS_POST_STAGING;
+        Topic topic = Constants.TOPIC_DRIVERS_POST_STAGING;
+        String warnings = "";
+        if (request.destinationTopic != null) {
+            if (request.destinationTopic.equals(Constants.TOPIC_STONE_TEST_TOPIC)) {
+                topic = request.destinationTopic;
+            } else {
+                if ((Long) request.destTopicJsonMap.get("category_id") == Constants.DRIVER_DELIVERIES_CATEGORY) {
+                    topic = request.destinationTopic;
+                } else {
+                    warnings = DEST_TOPIC_NOT_IN_DRIVER_DELIVERIES;
+                }
+            }
+        }
```

Comment:
I need the topic and the warning string, I don't see a way to save more than a couple lines, unless I make a kind of arbitrary object to return that holds both. Thoughts?

Reply:
Nothing useful. So, never mind.

For the not useful suggestion bucket, I would say if we an invalid dest topic was a failure instead of a warning, this could turn into a getDestinationTopic() method which returns a Topic or throws.  But that isn't the requirement.

Reply:
Got it, makes sense.